metadatastore/commands.py

```diff
@@ -839,6 +841,11 @@ def _transform_data(data, timestamps):
     to storage format:
         {'data': {<key>: (<value>, <timestamp>)}.
     """
+    # shallow copy so we can do some norrmalization
+    data = dict(data)
```

Comment:
isn't @stuwilkins doing something where the data and timestamps are both numpy arrays?  Does it makes sense to do the same thing for `timestamps` as you are doing for `data`?